presto-main/src/main/java/com/facebook/presto/cost/StatsCalculator.java

```diff
@@ -41,11 +41,15 @@ PlanNodeStatsEstimate calculateStats(
      * with the root PlanNode, so StatsCalculator can prepare for future `calculateStats` calls.
      *
      * @param root Root of plan tree for query
+     * @return true if registration succeeds, otherwise return false
      */
-    default void registerPlan(
+    default boolean registerPlan(
```

Comment:
if false returned, we will skip adding stats equivalent node in HBO marking optimizer.